### ISSUE TITLE
Add burst detection method: amplitude threshold computed from background fit

### ISSUE THREAD

**Issue by srcole** [voytekresearch/neurodsp#47]
As opposed to choosing the amplitude threshold for burst detection off the variance of the amplitude envelope, some promising methods fit a line to the power spectrum to determine the baseline amplitude level, if no oscillation is present. We should implement this in our burst module.

2 examples are:
* Better OSCillation Detector (BOSC)
    * https://www.sciencedirect.com/science/article/pii/S1053811910011614?via%3Dihub 
* Multiple Oscillations Detection ALgorithm (MODAL)
    * https://www.biorxiv.org/content/early/2017/10/12/202374 
    * http://www.andrewwatrous.com/uploads/7/8/5/9/78599468/modal.m

**Comment by rdgao** [comment]
I've been toying with the idea of fitting two distributions onto the amplitude histogram: one exponential for baseline noise activity when there is no oscillation (as per SCV) and one Gaussian (or log-normal) for periods of bursting. 

From the fitted distributions, you compute the probability of each time point falling into one or the other class, and you can either have a softmax threshold for classification or just return the probability of it being a bursting period.

**Comment by srcole** [comment]
Note that merged code separates BOSC into a separate burst detection function because the hyperparameters of this algorithm are very different than the two-threshold approach implemented in the first burst_detect function.
